lib/string.js

```diff
@@ -72,6 +72,16 @@ internals.String.prototype._base = function (value, state, options) {
         }
     }
 
+    if (typeof value === 'string' &&
+        this._inner.replacements &&
+        options.convert) {
```

Comment:
To me, the string should be left untouched unless the `convert` flag was specified (at validation time)

True that `replace` *IMPLIES* a modification (does not execute any test) but when it is paired with `regex`, it makes for some interesting constructs. For example:

```javascript
var schema = joi.string()
                .regex(/^[^x]*$/)
                .replace(/x/, 'y')
                .trim();

function validateOrNormalize(input, loose) {

  var result = joi.validate(input, schema, { convert: false } );
  if (result.error) {
    console.log("WARNING", result.error.details);
    if (loose) {
      return joi.validate(input, schema, { convert: true }).value;
    } else {
      throw result.error;
    }
  } else {
    return  input;
  }
}
```

Dunno, food for thought...